src/vs/workbench/contrib/editSessions/browser/editSessionsWorkbenchService.ts

```diff
@@ -173,30 +177,45 @@ export class EditSessionsWorkbenchService extends Disposable implements IEditSes
 			return true;
 		}
 
+		const authenticationSession = await this.getAuthenticationSession();
+		if (authenticationSession !== undefined) {
+			this.#authenticationInfo = authenticationSession;
+			this.storeClient.setAuthToken(authenticationSession.token, authenticationSession.providerId);
+		}
+
+		return authenticationSession !== undefined;
+	}
+
+	private async getAuthenticationSession() {
 		// If the user signed in previously and the session is still available, reuse that without prompting the user again
-		const existingSessionId = this.existingSessionId;
-		if (existingSessionId) {
-			this.logService.trace(`Searching for existing authentication session with ID ${existingSessionId}`);
-			const existing = await this.getExistingSession();
-			if (existing !== undefined) {
-				this.logService.trace(`Found existing authentication session with ID ${existingSessionId}`);
-				this.#authenticationInfo = { sessionId: existing.session.id, token: existing.session.idToken ?? existing.session.accessToken, providerId: existing.session.providerId };
-				this.storeClient.setAuthToken(this.#authenticationInfo.token, this.#authenticationInfo.providerId);
-				return true;
+		if (this.existingSessionId) {
+			this.logService.trace(`Searching for existing authentication session with ID ${this.existingSessionId}`);
+			const existingSession = await this.getExistingSession();
+			if (existingSession) {
+				this.logService.trace(`Found existing authentication session with ID ${existingSession.session.id}`);
+				return { sessionId: existingSession.session.id, token: existingSession.session.idToken ?? existingSession.session.accessToken, providerId: existingSession.session.providerId };
+			}
+		}
+
+		// If settings sync is already enabled, avoid asking again to authenticate
+		if (this.userDataSyncEnablementService.isEnabled()) {
+			this.logService.trace(`Reusing user data sync enablement`);
+			const authenticationSessionInfo = await getCurrentAuthenticationSessionInfo(this.credentialsService, this.productService);
+			if (authenticationSessionInfo !== undefined) {
```

Comment:
again - just stylistic, but `!!authenticationSessionInfo` or `authenticationSessionInfo` would work here

Reply:
I personally try to use explicit checks for `undefined`, because in the past the following has gotten me into trouble:

```ts
const foo: boolean = bar();
if (foo) {
	// do something
}
```

and if the return type of `bar()` later changes to be an object, `if (foo)` now does not do what I want.